GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+                newServicePrincpal.DisplayName = newApp.DisplayName;
+                newServicePrincpal.AccountEnabled = true;
+                newServicePrincpal.AppId = newApp.AppId;
+                try
+                {
+                    await client.ServicePrincipals.AddServicePrincipalAsync(newServicePrincpal);
+                    Console.WriteLine("New Service Principal created: " + newServicePrincpal.DisplayName);
+                }
+                catch (Exception e)
+                {
+                    Program.WriteError("\nError Creating Service Principal: {0} {1}", e.Message,
+                        e.InnerException != null ? e.InnerException.Message : "");
+                }
+            }
+            return newServicePrincpal;
+
+            #endregion
+        }
+
+        private static async Task<ExtensionProperty> CreateSchemaExtensions(IActiveDirectoryClient client, 
+            Application newApp, 
+            string extName)
+        {
+            #region Create an Extension Property
+
+            // **************************************************************************************************
+            // Create a new extension property - to extend the user entity
+            // This is accomplished by declaring the extension property through an application object
+            // **************************************************************************************************
+            if (newApp.ObjectId != null)
+            {
+                ExtensionProperty linkedInUserId = new ExtensionProperty
+                {
+                    Name = extName,
+                    DataType = "String",
+                    TargetObjects = {"User"}
+                };
+                try
+                {
+                    // firstly, let's write out all the existing cloud extension properties in the tenant
+                    IEnumerable<IExtensionProperty> allExts = await client.GetAvailableExtensionPropertiesAsync(false);
+                    foreach (ExtensionProperty ext in allExts)
+                    {
+                        Console.WriteLine("\nExtension: {0}", ext.Name);
+                    }
+                    newApp.ExtensionProperties.Add(linkedInUserId);
+                    await newApp.UpdateAsync();
+                    Console.WriteLine("\nUser object extended successfully with extension: {0}.", linkedInUserId.Name);
+                }
+                catch (Exception e)
+                {
+                    Program.WriteError("\nError extending the user object {0} {1}", e.Message,
+                        e.InnerException != null ? e.InnerException.Message : "");
+                }
+                return linkedInUserId;
+
+                #endregion
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        private static async Task ManipulateExtensionProperty(Application newApp, string extName, User newUser)
```

Comment:
>ManipulateExtensionProperty [](start = 34, length = 27)

This will try to execute even when the extension property creation failed resulting in the failure here too.